Add fraud proof to Zero-Knowledge Glossary (#357)
Consensys/doc.zk-evm: docs/zero-knowledge-glossary/index.mdx

## F

### Fraud proof

> A fraud proof (fault proof) is the evidence provided by a verifier to question the state of a transaction. Depending on how it's executed, this proof scrutinizes a section of the transactions and endeavors to confirm their validity.

Reviewer: ```suggestion
In optimistic rollups, fraud proofs (fault proofs) are the evidence a verifier provides to question the validity of a batch of transactions. Depending on how they are executed, these proofs scrutinize a section of transactions and endeavor to confirm their validity.
```

Author: added
